### PR TITLE
Prevent Tags in ChatBubbles and Comms Console

### LINKED ISSUE #8112: Improper sanitization in chat bubbles.
### Environment
- Game Version: 22010322
- OS Version: n/a

### Detailed description of bug

Chat bubbles are not sanitised for TMP/RichText Tags. Resulting in people being able to make *hilariously* large text and other naughty stuff inside chat bubbles.

### Steps to reproduce

1.  Enable chat bubbles
2.  Include TMP tags inside a message
3. Watch as they are honored inside your chat bubble, but stripped from the speech.

### PR DESCRIPTION
**Pull request by ThatDan123**
-Prevent Tags in ChatBubbles and Comms Console

Fixes #8112

CL: [Fix] fixed an issue with chat bubbles and the comms console not being fully sanitized of richtext tags.

### CODE REVIEW

**Review by craftxbox — CHANGES_REQUESTED** [review]
AI has a dedicated, separate button for calling shuttle that isnt in the comms gui. Aswell, confirm this won't affect admins through the admin panels?

**Comment by ThatDan123** [comment]
> AI has a dedicated, separate button for calling shuttle that isnt in the comms gui. Aswell, confirm this won't affect admins through the admin panels?

Cool, stripped Ai tags too.

This won't affect the admin stuff.

**Comment by ThatDan123** [comment]
Codacy is complaining about other stuff so can be ignored.

File: UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs [hunk 1]
```diff
@@ -342,7 +342,7 @@ private static string StripAll(string input)
 		return output;
 	}
 
-	private static string StripTags(string input)
+	public static string StripTags(string input)
 	{
 		//Regex - find "<" followed by any number of not ">" and ending in ">". Matches any HTML tags.
 		Regex rx = new Regex("[<][^>]+[>]");
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs:348]: just saying this stuff is pretty slow when I did profile it

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs:348]: The only ones server side i added were the ones for the comms console. The chat bubble will strip client side

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs:348]: I'm not sure if theres a better way to do it?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs:348]: I think there is built-in replace that might work well <  with ""

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs:348]: It also gets rid of the stuff in between the <>

Might still want that? Not sure

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs:348]: yeah, it is stripping the entire tag so I think regex is needed here. Another way would be finding the "<" character taking note of its index then find the ">" character and make a substring but that's is way too prone to fail in my opinion.

**Review by corp-0 — APPROVED** [review]

File: UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Chat/Chat.Process.cs:348]: but it's fine currently

**Comment by NoooneyDude** [comment]
~You can put the text content in `<noparse> </noparse>`. No stripping needed.~

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**